UserExperience_Plugin_Jquery.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * File: UserExperience_Plugin_Jquery.php
+ *
+ * @since 0.14.4
+ *
+ * @package W3TC
+ */
+
+namespace W3TC;
+
+/**
+ * Class: UserExperience_Plugin_Jquery
+ *
+ * @since 0.14.4
+ */
+class UserExperience_Plugin_Jquery {
+	/**
+	 * Config class object.
+	 *
+	 * @since 0.14.4
+	 *
+	 * @var Config
+	 */
+	private $_config = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 0.14.4
+	 */
+	public function __construct() {
+		$this->_config = Dispatcher::config();
+	}
+
+	/**
+	 * Run.
+	 *
+	 * @since 0.14.4
+	 */
+	public function run() {
+		// Disable jquery-migrate on the front-end, if configured.
+		if ( ! is_admin() && $this->_config->get_boolean( 'jquerymigrate.disabled' ) ) {
+			add_action( 'wp_default_scripts', array( $this, 'disable_jquery_migrate' ) );
+		}
+	}
+
+	/**
+	 * Disable jquery-migrate.
```

Comment:
jQuery migrate is not removed if I enqueue it by itself.

```
add_action(
    'wp_enqueue_scripts',
    function() {
	    wp_enqueue_script( 'jquery-migrate' );
    },
    100
);
```

Current approach only works if it's registered as a dependency.